### ISSUE TITLE
ClientSocketWs.Connect() accepts timeout param which is never used

### ISSUE THREAD

**Issue by Vennril** [alvyxaz/barebones-masterserver#88]
In fact, I don't think WebSocket connections handle connection timeouts at all.

**Comment by alvyxaz** [comment]
Yeah, you're right - apparently I forgot to handle the timeout when I thought I'd do that at a "higher level". I couldn't decide if it was worth handling disconnections on the socket implementation level, or leave it to  whatever's using the connection to implement it

And thanks a lot for reporting all of the other issues! It's some high quality QA, haha
